src/facilities-context/FacilitiesContext.tsx

```diff
@@ -259,7 +283,10 @@ export const FacilitiesProvider: React.FC<{ children: React.ReactNode }> = ({
     const facilities = Object.values({ ...state.facilities });
     if (facilities.length) {
       facilities.forEach((facility) => {
-        if (!state.rtData.hasOwnProperty(facility.id)) {
+        if (
+          !state.rtData.hasOwnProperty(facility.id) &&
+          hasCases(facility.modelInputs)
```

Comment:
i'm not so sure about this ... I see why it works here but I am a bit concerned that it really depends on a side effect of the way we are creating these facilities. I think maybe if we can find a way to make those updates imperatively in such a way that we prevent the "intermediate" facility objects from showing up in the context (as we purport to do on L275) then we wouldn't have to do this anymore. 

I had an idea about that .... 

what if we made a new action function that combined most of the [handleSave](https://github.com/Recidiviz/covid19-dashboard/blob/master/src/page-multi-facility/ReferenceDataModal/SyncNoUserFacilities.tsx#L249) function in `SyncNoUserFacilities` with the new `updateFacilities` function you added above, in such a way that it waits to update the context until after the reference facilities are fetched? (we are going to have to factor that out anyway because there will eventually be two modals from which you can create new facilities from reference data, Neehar and I are working on the second one now)?

We can defer that if you think it's too much scope increase for this PR (I wouldn't argue with that) but wonder what you think of the idea anyway, or whether you have a better one

Reply:
I like the idea of creating the composite facilities right away in the handleSave of the modal, I think that would remove the need for this `hasCases` check here and then like you said we wouldn't have these "intermediate" facility objects. I had originally thought about doing that, but this seemed like the quicker fix at the time, though I can see that it's hidden in here and might cause other side effects. I'll work on updating the handleSave of the modal so that it creates the new facilities and updates the context with the composite facilities. Let me know if that sounds right to you or if I misunderstood anything!

Reply:
yeah that sounds good! thanks